packages/features/portfolio/src/containers/portfolio-details/vault-holdings-detail.tsx

```diff
@@ -0,0 +1,279 @@
+          </TokenContainer>
+          <Box
+            sx={{
+              display: 'flex',
+              flexDirection: 'column',
+              alignItems: 'flex-end',
+              justifyContent: 'flex-start',
+            }}
+          >
+            <Header variant="h2" sx={{ marginBottom: 0 }}>
+              {value.marketAPY}
+            </Header>
+          </Box>
+        </HeaderContainer>
+
+        <ContentContainer>
+          <Title variant="h3">
+            <FormattedMessage
+              id="earningsSummary"
+              defaultMessage="Earnings Summary"
+            />
+          </Title>
+          <Row>
+            <Label>
+              <FormattedMessage id="amountPaid" defaultMessage="Amount Paid" />
+            </Label>
+            <Column>
+              {typeof value.amountPaid === 'string' ? (
+                <Value>{value.amountPaid}</Value>
+              ) : (
+                <>
+                  <Value>{value.amountPaid.data?.[0]?.displayValue}</Value>
+                  <Caption>{value.amountPaid.data?.[1]?.displayValue}</Caption>
+                </>
+              )}
+            </Column>
+          </Row>
+
+          <Row>
+            <Label>
+              <FormattedMessage
+                id="presentValue"
+                defaultMessage="Present Value"
+              />
+            </Label>
+            <Column>
+              {typeof value.presentValue === 'string' ? (
+                <Value>{value.presentValue}</Value>
+              ) : (
+                <>
+                  <Value>{value.presentValue.data?.[0]?.displayValue}</Value>
+                  <Caption>
+                    {value.presentValue.data?.[1]?.displayValue}
+                  </Caption>
+                </>
+              )}
+            </Column>
+          </Row>
+
+          <Row>
+            <Label>
+              <FormattedMessage id="earnings" defaultMessage="Earnings" />
+            </Label>
+            <Column>
+              {typeof value.totalEarnings === 'string' ? (
+                <Value sx={{ color: theme.palette.primary.main }}>
+                  {value.totalEarnings}
+                </Value>
+              ) : (
+                <>
+                  <Value sx={{ color: theme.palette.primary.main }}>
+                    {value.totalEarnings.data?.[0]?.displayValue}
+                  </Value>
+                  <Caption>
+                    {value.totalEarnings.data?.[1]?.displayValue}
+                  </Caption>
+                </>
+              )}
+            </Column>
+          </Row>
+        </ContentContainer>
+
+        <ContentContainer>
+          <Title variant="h3">
+            <FormattedMessage id="details" defaultMessage="Details" />
+          </Title>
+          {value.actionRow.subRowData.map((row, index) => (
+            <Row key={index}>
+              <Label>{row.label}</Label>
+              <Column>
+                <Value>{row.value}</Value>
```

Comment:
Some components might be a ReactNode instead of a string and the alignment looks wrong on Leverage Ratio in this screenshot

![image](https://github.com/user-attachments/assets/a8d931e7-41f4-406f-a625-aa7ff1389354)